packages/amesos2/cmake/Dependencies.cmake

```diff
@@ -8,6 +8,6 @@ SET(TEST_REQUIRED_DEP_PACKAGES)
 SET(TEST_OPTIONAL_DEP_PACKAGES ShyLU_NodeBasker ShyLU_NodeTacho Kokkos TrilinosSS)
 # SET(LIB_REQUIRED_DEP_TPLS SuperLU)
 SET(LIB_REQUIRED_DEP_TPLS )
-SET(LIB_OPTIONAL_DEP_TPLS MPI SuperLU SuperLUMT SuperLUDist UMFPACK PARDISO_MKL ParMETIS METIS Cholmod MUMPS)
+SET(LIB_OPTIONAL_DEP_TPLS MPI SuperLU SuperLUMT SuperLUDist UMFPACK PARDISO_MKL ParMETIS METIS Cholmod MUMPS CUSPARSE CUSOLVER)
```

Comment:
This is the key.  You just need to tell TriBITS that `CUSPARSE` and `CUSOLVER` are optional TPLs for the `Amesos2` package and TriBITS takes care of defining and setting the following vars before processing the file `packages/amesos2/CMakeLists.txt`:

* `Amesos2_ENABLE_CUSPARSE`
* `HAVE_AMESOS2_CUSPARSE`
* `Amesos2_ENABLE_CUSOLVER`
* `HAVE_AMESOS2_CUSOLVER`

and likewise for all of these other optional TPLs.